src/test/java/com/thirdcc/webapp/web/rest/EventActivityResourceIT.java

```diff
@@ -275,25 +448,217 @@ public void updateNonExistingEventActivity() throws Exception {
+
+        restEventActivityMockMvc.perform(put("/api/event-activities")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(eventActivityDTO)))
+            .andExpect(status().isBadRequest());
+
+        List<EventActivity> eventActivityList = eventActivityRepository.findAll();
+        assertThat(eventActivityList).hasSize(databaseSizeBeforeUpdate);
+    }
+
+    @Test
+    public void updateEventActivity_WithStartDateEarlierThanEventStartDate_ShouldThrow400() throws Exception {
+        event.setStartDate(Instant.now().plus(1, ChronoUnit.DAYS));
+        Event savedEvent = initEventDB();
+        EventActivity savedEventActivity = initEventActivityDB(savedEvent);
+
+        int databaseSizeBeforeCreate = eventActivityRepository.findAll().size();
+
+        EventActivityDTO eventActivityDTO = createUpdateEventActivityDTO();
+        eventActivityDTO.setId(savedEventActivity.getId());
+        eventActivityDTO.setStartDate(savedEvent.getStartDate().minus(1, ChronoUnit.SECONDS));
+
+        assertThat(savedEvent.getStartDate()).isAfter(Instant.now()); //for more accurate testing
+        assertThat(eventActivityDTO.getStartDate()).isBefore(savedEvent.getStartDate());
+
+        restEventActivityMockMvc.perform(put("/api/event-activities")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(eventActivityDTO)))
+            .andExpect(status().isBadRequest());
+
+        List<EventActivity> eventActivityList = eventActivityRepository.findAll();
+        assertThat(eventActivityList).hasSize(databaseSizeBeforeCreate);
+    }
+
+    @Test
+    public void updateEventActivity_WithStartDateLaterThanEventEndDate_ShouldThrow400() throws Exception {
+        Event savedEvent = initEventDB();
+        EventActivity savedEventActivity = initEventActivityDB(savedEvent);
+
+        int databaseSizeBeforeCreate = eventActivityRepository.findAll().size();
+
+        EventActivityDTO eventActivityDTO = createUpdateEventActivityDTO();
+        eventActivityDTO.setId(savedEventActivity.getId());
+        eventActivityDTO.setStartDate(savedEvent.getEndDate().plus(1, ChronoUnit.SECONDS));
+
+        assertThat(eventActivityDTO.getStartDate()).isAfter(savedEvent.getEndDate());
+
+        restEventActivityMockMvc.perform(put("/api/event-activities")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(eventActivityDTO)))
+            .andExpect(status().isBadRequest());
+
+        List<EventActivity> eventActivityList = eventActivityRepository.findAll();
+        assertThat(eventActivityList).hasSize(databaseSizeBeforeCreate);
+    }
+
+
+    @Test
     public void deleteEventActivity() throws Exception {
-        // Initialize the database
-        eventActivityRepository.saveAndFlush(eventActivity);
+        event.setEndDate(Instant.now().minus(1, ChronoUnit.DAYS));
```

Comment:
why set event to be ended? endDate < today

Reply:
Good Catch 👍 
this is a mistake. The user should be able to delete eventActivities of event that is **NOT** ended. 
_which in this case, it is the opposite, the user can only delete eventActivites if the event is ended_ 🤦‍♂️

fixed in 32c8feeefe1ca6221761c8748a79059953d83cda